### PR TITLE
(task) tdd mock, stub, spy; js type

### PR DESCRIPTION
**Pull request by dan-h-ch**

### CODE REVIEW

File: answers/general/javascript-type.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+## Strong vs weak
+
+Related to static vs dynamic, strong vs weak related to if a pointer can be reassigned to a value of a different type. In the case of Javascript, this is possile. We are able to do somethin like below.
+
+```Javascript
+var a = "string"
+a = 3 // this is allowed
+```
+
+Additionally, [type coercion](what-is-type-coercion.md) can occured because Javascript is weekly typed. If it were not, type coercion would not be possible.
```
Comment (therobinkim) [inline answers/general/javascript-type.md:19]: Javascript is `weakly` typed

File: answers/bestpractices/tdd-spy-mock-stub.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+A spy is typically used to gather information about function calls. In essence, a spy can hijack a function and add additional properties that allow you observe other things about that function. For example, we can determine how many times a function was called and validate that against any expected behavior. It's possible we want to make sure an function for an AJAX call we implemented only runs once, we could use a spy to determine that.
+
+### What's a stub?
+
+A stub is a function that replaces another function completedly. Instead of only hijacking and adding properties, it completed replaces the function. Stubs are useful when you want to guarantee a specific response. For example, our AJAX call needs to get a specific response, but it's possible the endpoint is not up and running, in this case we could use a stub.
+
+### What's a mock?
+
+Mocks are very similat to subs, but are primarily useful if you want to sub more than on function on a given object. Instead of replacing functions, you can mock and entire object.
```
Comment (therobinkim) [inline answers/bestpractices/tdd-spy-mock-stub.md:23]: Mocks are very `similar` to `stubs`,...
...you can mock `an` entire object.

File: README.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ Some suggested answers along with relevant resources are posted in the answers f
 * [What is the difference between == and ===.](answers/general/double-equal-vs-triple-equal.md)
 * [What is the difference between call, apply, and bind?](answers/general/call-apply-bind.md)
 * [What is the difference between encryption and hashing?](answers/general/encryption-vs-hashing.md)
+* [Is Javascript static or dnyamic? Is it strong or weak type?](answers/general/javascript-type.md)
```
Comment (therobinkim) [inline README.md:35]: Is `JavaScript` static or `dynamic`? Is it `strongly` or `weakly` typed?

File: README.md [hunk 4]
```diff
@@ -75,6 +77,7 @@ Some suggested answers along with relevant resources are posted in the answers f
 * [Why is eval() considered evil and dangerous?](answers/bestpractices/eval-dangerous.md)
 * [What is meant by 'repaint and reflow'? How does it affect your application?](answers/bestpractices/repaint-reflow.md)
 * [What is currying?](answers/bestpractices/what-is-currying.md)
+* [In TDD what is difference between a spy, mock, and stub?](answers/bestpractices/tdd-spy-mock-stub.md)
```
Comment (therobinkim) [inline README.md:80]: what is `the` difference

File: answers/bestpractices/tdd-spy-mock-stub.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+There is some vagueness to this question, but it's used as a basis if you've actually understand TDD at a deeper level. Most of the answers in this readme are based on usage with Sinon. Terminology/use may vary with a different testing framework.
+
+### What's a spy?
+
+A spy is typically used to gather information about function calls. In essence, a spy can hijack a function and add additional properties that allow you observe other things about that function. For example, we can determine how many times a function was called and validate that against any expected behavior. It's possible we want to make sure an function for an AJAX call we implemented only runs once, we could use a spy to determine that.
+
+### What's a stub?
+
+A stub is a function that replaces another function completedly. Instead of only hijacking and adding properties, it completed replaces the function. Stubs are useful when you want to guarantee a specific response. For example, our AJAX call needs to get a specific response, but it's possible the endpoint is not up and running, in this case we could use a stub.
```
Comment (therobinkim) [inline answers/bestpractices/tdd-spy-mock-stub.md:19]: `completely`
`completely`

**Review by therobinkim — COMMENTED** [review]
spelling/grammar suggestions

**Comment by dan-h-ch** [comment]
Thanks for feedback.

**Pull request merged**